src/panels/config/voice-assistants/assist/ha-config-voice-assistants-assist-devices.ts

```diff
@@ -0,0 +1,148 @@
+import { LitElement, PropertyValues, html } from "lit";
+import memoizeOne from "memoize-one";
+import { customElement, property, state } from "lit/decorators";
+import "../../../../layouts/hass-subpage";
+import "../../../../components/data-table/ha-data-table";
+import type { DataTableColumnContainer } from "../../../../components/data-table/ha-data-table";
+import { HomeAssistant } from "../../../../types";
+import {
+  AssistDevice,
+  AssistPipeline,
+  listAssistDevices,
+  listAssistPipelines,
+} from "../../../../data/assist_pipeline";
+import { computeDeviceName } from "../../../../data/device_registry";
+import { navigate } from "../../../../common/navigate";
+
+interface AssistDeviceExtra {
+  pipeline: string | undefined;
+  last_used: string | undefined;
+}
+
+@customElement("ha-config-voice-assistants-assist-devices")
+class AssistDevicesPage extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ type: Boolean }) public narrow!: boolean;
+
+  @state() private _pipelines: Record<string, AssistPipeline> = {};
+
+  @state() private _preferred: string | null = null;
+
+  @state() private _devices?: (AssistDevice | AssistDeviceExtra)[];
+
+  private _columns = memoizeOne(
+    (
+      hass: HomeAssistant,
+      pipelines: Record<string, AssistPipeline>,
+      preferred: string | null
+    ): DataTableColumnContainer => {
+      const columns: DataTableColumnContainer<AssistDevice> = {
+        name: {
+          title: hass.localize(
+            "ui.panel.config.voice_assistants.assistants.pipeline.devices.device"
+          ),
+          width: "50%",
+          filterable: true,
+          sortable: true,
+          template: (assistDevice) =>
+            computeDeviceName(hass.devices[assistDevice.device_id], hass),
```

Comment:
The drawback of doing this in the template is that it can not be used for filtering